main.cpp

```diff
@@ -62,23 +60,9 @@ struct FloatType
     FloatType& subtract(float rhs );
     FloatType& multiply(float rhs );
     FloatType& divide(float rhs );
-    // 4 functions taking a FloatType as input
-    FloatType& add(const FloatType& rhsFt );
-    FloatType& subtract(const FloatType& rhsFt );
-    FloatType& multiply(const FloatType& rhsFt );
-    FloatType& divide(const FloatType& rhsFt );
-    // 4 functions taking a DoubleType as input
-    FloatType& add(const DoubleType& rhsDt );
-    FloatType& subtract(const DoubleType& rhsDt );
-    FloatType& multiply(const DoubleType& rhsDt );
-    FloatType& divide(const DoubleType& rhsDt );
-    // 4 functions taking an int as input
-    FloatType& add(const IntType& rhsIt );
-    FloatType& subtract(const IntType& rhsIt );
-    FloatType& multiply(const IntType& rhsIt );
-    FloatType& divide(const IntType& rhsIt );
+    // convert UDT to primitive type
+    operator float();
```

Comment:
It's easiest to just implement these in-class
- [ ] move conversion function implementation to inside of class.